openpype/hosts/maya/api/pipeline.py

```diff
@@ -426,6 +468,49 @@ def on_before_save():
     return lib.validate_fps()
 
 
+def after_file_open():
+    """Check if there is a user opening the file"""
+    log.info("Running callback on checking the lock file...")
+
+    # add the lock file when opening the file
+    filepath = current_file()
+
+    if not is_workfile_locked(filepath):
+        create_workfile_lock(filepath)
+
+    else:
+        username = get_username(filepath)
+        reminder = cmds.window(title="Reminder", width=400, height=30)
+        cmds.columnLayout(adjustableColumn=True)
+        cmds.separator()
+        cmds.columnLayout(adjustableColumn=True)
+        comment = " %s is working the same workfile!" % username
+        cmds.text(comment, align='center')
+        cmds.text(vis=False)
+        cmds.rowColumnLayout(numberOfColumns=3,
+                             columnWidth=[(1, 300), (2, 100)],
+                             columnSpacing=[(2, 10)])
+        cmds.separator(vis=False)
+        quit_command = "cmds.quit(force=True);cmds.deleteUI('%s')" % reminder
+        cmds.button(label='Ok', command=quit_command)
```

Comment:
Better maybe to not quit Maya but instead "close" the file.

Also feel like this UI logic is better implemented in Qt to stick with OpenPype styling and also potentially introduce this logic in another host. I feel like this most of this logic should actually be in `openpype.pipeline.lock_workfile` too.

Reply:
> Better maybe to not quit Maya but instead "close" the file? (E.g. new scene?)

True

> Also feel like this UI logic is better implemented in Qt to stick with OpenPype styling and also potentially introduce this logic in another host. I feel like this most of this logic should actually be in openpype.pipeline.lock_workfile too.

I agree but I'm not sure if @moonyuet should do it, and it should be done in this PR. We do this because of single request and spending so much time because of single request without knowing if it actually be used is waste of work time for us. But the lock should be in settings and disabled by default.